Update TCP/UDP Settings to Reside in the new network.lua file

jmcdaniel83 opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

This utilizes the existing new LUA settings files for the TCP/UDP settings.  The settings from the two previous files `src/common/packet_tcp.conf` and `src/common/packet_udp.conf` have been removed in favor of the new settings setup.  `src/common/socket.cpp` has been updated to leverage the new `settings.h` file to pull specific setting parameters to make this process much cleaner and consistent throughout the code and makes the deployment much easier.  The only potential issue is that currently we no longer would be able to support imports from other TCP/UDP setting files.

I've also added some comments in `src/map/zone.cpp` in order to identify which navmesh files are not properly loading.

## Steps to test these changes

I've gone through the newly updated `settings/default/network.lua` file and validated that modifying the options is affecting who can log in to the server (allowed or denied) and have been able to toggle debug messages off and on.  Have also tested turning off the IP rules entirely and made sure that was working as well.

jmcdaniel83 committed:
Moving all packet configuration into the network.lua file
 + to differentiate between the two sets I've added either UDP_ or TCP_ to the key name
 + all items have been provided along with the comments specified
 + for IP rules, the TCP_ALLOW, TCP_DENY keys have been updated to contain lists
   + default is empty string (no-op) in the code during setup
   + each entry must be separated with a 'comma' character
  - removed old configuration files (no longer referenced)
  + all settings now consistent across project

jmcdaniel83 committed:
updating dockerfile for readability
 + format changes, no logical changes

jmcdaniel83 committed:
Adding use of the new settings get<T>() method for all configuration items.
 + added two new booleans (udp_debug, and tcp_debug) to differentiate between them (not yet used)
 + connect_check_() method for loops updated to leverage the range for loop to simplify the code and for more readabilty using the new entry element each loop
 + access_ipmask() updated to allow for emtpy strings which equate to a no-op
 + adding new methods to leverage the new settings get<T> method
   + get_access_list() takes the parsed access list string "entry,entry,..." and provides a list of AccessControl objects that have been successfully parsed, all issues with the entries will be shown in an error message.
   + socket_udp_setup() method added to setup our UDP socket (only fetches the debug setting right now)
   + socket_tcp_setup() method added to setup the TCP socket settings. Much simpler with the new get<T>() method.  Only issue I had was with the list of allow and deny IP addresses as we would need to extend the get<T>() method to allow LUA lists { "a", "b" }.  All other properties are straight forward using the get<T>() method.  The only option that is not carried over is the "import" command (not sure this is used anymore).
   - socket_config_read() now deprecated
 + socket_init_tcp() method updated to leverage new TCP setup method, some formatting for readability
 + socket_init_udp() levearging new UDP setup method

jmcdaniel83 committed:
Moved handling of empty strings for access/deny strings to where the access list is collected
 + moved handling of empty strings from access_ipmask() to get_access_list()
 + adding more log information of what was parsed in the new socket_tcp_setup() function

jmcdaniel83 committed:
adding some helpful critical messages when nav meshes can not be loaded during boot

jmcdaniel83 committed:
Adding all options for TCP_ORDER setting, more examples for ALLOW and DENY

zach2good commented:
Cool! I was too scared to rip these files apart during the settings refactor. Nice work 👍

Xaver-DaRed commented:
Gonna need a rebase to solve conflicts

jmcdaniel83 committed:
Reverting dockerfile back to original to make merge easier

jmcdaniel83 committed:
Merge branch 'base' of https://github.com/LandSandBoat/server into socket-config-updates fork

jmcdaniel83 commented:
have merged from `base` branch into my fork to complete pull request

zach2good commented:
Could you rebase onto the latest `base`  removing the "revert" commit, instead of merging? That way you'll get to keep your commits. Otherwise, we can just squash when we merge.

jmcdaniel83 commented:
Just to make sure that I don't mess up anything, do you want me to do the following on my fork?

Before the `[8d273f5]` commit, perform the following
```
git rebase [what]
```

We can squash it for the moment since I don't want to mess up anything but, just for future reference, just need to know how to properly rebase from a fork into the main `base` branch.

jmcdaniel83 committed:
Corrected proper formating to the file

cocosolos commented:
If you ever just want to look back at something like the last 9 commits and drop/edit/squash/rearrange/whatever some of them, you can do `git rebase -i HEAD~9` and it'll bring up your text editor and you can change what you need to then force push. It gets a little harder when you have other branches merging like here though. If you want to keep your commits in order and include new commits from upstream you can use `git rebase --onto <new_parent> <old_parent>` and it'll replay your commits on top of the new parent instead of merging them in and creating a mixed history. This is a little more complicated IMO and you'll probably want to look up [better instructions](https://git-scm.com/docs/git-rebase) before attempting that.

zach2good commented:
If you want to be a filthy casual like me, I find the VSCode git workflow to be pretty easy to use also

zach2good commented on settings/default/network.lua:
```diff
@@ -31,4 +31,79 @@ xi.settings.network =
     ZMQ_IP   = "127.0.0.1",
     ZMQ_PORT = 54003,
+
+    -- ===========================
```
Would you mind adding an additional comment block around here, saying that the "below settings are advanced options and it is very unlikely that users will need to modify them", or something to that effect?

Once these are in plainer view of regular users, I have no doubt that people will start to fiddle with them and start coming to us with all sorts of evil networking issues

jmcdaniel83 replied:
That's a good call

jmcdaniel83 committed:
Have added a helpful note to signify the settings that should be left alone
 + most options can render your instance unusable unless you know what you are doing.

zach2good reviewed (approved):
Great, thanks!
